src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs

```diff
@@ -220,6 +227,23 @@ protected void AddMapInternal(string collection, string subCollecction, Indexing
             funcs.Add(map);
         }
 
+        internal void CheckDepthOfStackInOutputMap(IndexDefinition indexMetadata, DocumentDatabase documentDatabase)
+        {
+            var performanceHintConfig = documentDatabase.Configuration.PerformanceHints;
+            if (StackSizeInSelectClause > performanceHintConfig.MaxDepthOfRecursionInLinqSelect)
+            {
+                documentDatabase.NotificationCenter.Add(PerformanceHint.Create(
+                    documentDatabase.Name,
+                    $"Index '{indexMetadata.Name}' contains a lot of `let` clauses.",
+                    $"Each of the let clause is nesting projections and could potentially leads to StackoverflowException.",
+                    PerformanceHintType.Indexing,
+                    NotificationSeverity.Info,
+                    nameof(IndexCompiler)));
+                
+                Log.Info($"Index '{indexMetadata.Name}' contains a lot of `let` clauses.");
```

Comment:
1. Let's change it to Operations level (it's turned on by default). Info level is much more detailed and it's supposed to be enabled on demand if needed
2. Before doing the actual logging we always check if the logger itself is gonna do it so please add `if` as below:

```
if (Log.IsOperationsEnabled)
    Log.Operations(...)
```

3. I wonder if we shouldn't include `StackSizeInSelectClause` number here as well